src/Sentry.NLog/SentryTarget.cs

```diff
@@ -0,0 +1,249 @@
+        protected override void InitializeTarget()
+        {
+            base.InitializeTarget();
+
+            IncludeEventProperties = Options.SendEventPropertiesAsData;
+
+            if (Options.MarkNLogNamespaceAsExternal)
+            {
+                Options.AddInAppExclude("NLog");
+            }
+
+            // If the sdk is not there, set it on up.
+            if (Options.InitializeSdk && _sdkDisposable == null)
+            {
+                _sdkDisposable = SentrySdk.Init(Options);
+            }
+        }
+
+        /// <inheritdoc />
+        protected override void Write(LogEventInfo logEvent)
+        {
+            if (logEvent?.Message == null)
+            {
+                return;
+            }
+
+            var hub = _hubAccessor();
+
+            if (hub?.IsEnabled != true)
+            {
+                return;
+            }
+
+            var exception = logEvent.Exception;
+            var formatted = Layout.Render(logEvent);
+            var template = logEvent.Message;
+
+            var contextProps = GetAllProperties(logEvent);
+
+            var shouldOnlyLogExceptions = exception == null && Options.IgnoreEventsWithNoException;
+
+            if (logEvent.Level >= Options.MinimumEventLevel && !shouldOnlyLogExceptions)
+            {
+                var evt = new SentryEvent(exception)
+                {
+                    Sdk =
+                    {
+                        Name = Constants.SdkName,
+                        Version = NameAndVersion.Version
+                    },
+                    Message = null,
+                    LogEntry = new LogEntry
+                    {
+                        Formatted = formatted,
+                        Message = template
+                    },
+                    Level = logEvent.Level.ToSentryLevel(),
+                    Release = Options.Release,
+                    Environment = Options.Environment,
+                };
+
+                evt.Sdk.AddPackage(ProtocolPackageName, NameAndVersion.Version);
+
+                // Always apply any manually configured tags
+                evt.SetTags(GetTags(logEvent));
+
+                if (IncludeEventProperties)
+                {
+                    evt.SetExtras(contextProps);
+                }
+
+                if (Options.SendEventPropertiesAsTags)
+                {
+                    evt.SetTags(GetLoggingEventProperties(logEvent).MapKeys(a => a.ToString()));
+                }
+
+                hub.CaptureEvent(evt);
+            }
+
+            // Whether or not it was sent as event, add breadcrumb so the next event includes it
+            if (logEvent.Level >= Options.MinimumBreadcrumbLevel)
+            {
+                IDictionary<string, string> data = null;
+
+                // If this is true, an exception is being logged with no custom message
+                if (exception != null && !string.IsNullOrWhiteSpace(formatted) && logEvent.Message != "{0}")
```

Comment:
Amazing attention to detail here. Maybe one could write it as this:
```
                var message = string.IsNullOrWhiteSpace(formatted)
                    ? (exception?.Message ?? string.Empty)
                    : formatted;
               if (exception != null && !message.StartsWith(exception.Message))
               {
                    // Exception.Message won't be used as Breadcrumb message Avoid losing it by adding as data:
                    data = new Dictionary<string, string>
                        {
                            { "exception_message", exception.Message }
                        };
               }
```
Any reason you don't want to include `exception_type`. `exception.GetType()`.

Reply:
Thanks, and good idea on the optimization. And no, there was no reason I was purposefully excluding the exception type, and I do think it should be included, so I'll add that in.